File: app/src/main/java/com/doapps/dojoke/MainActivity.java
package com.doapps.dojoke;

import android.app.Activity;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.ScaleAnimation;
import android.widget.ImageView;

public class MainActivity extends Activity {
    private MediaPlayer daDumMp;
    private MediaPlayer cricket;
    private MediaPlayer aww;
    private MediaPlayer fakeLaugh;
    private ImageView laughImage;
    private ImageView awwImage;
    private ImageView cricketImage;
    private ImageView daDumTssImage;

    private static ScaleAnimation fade_out;
    private static ScaleAnimation fade_in;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Media players
        daDumMp = MediaPlayer.create(this, R.raw.badum);
        cricket = MediaPlayer.create(this, R.raw.cricket);
        aww = MediaPlayer.create(this, R.raw.aww);
        fakeLaugh = MediaPlayer.create(this, R.raw.laugh);

        // Images
        laughImage = (ImageView) findViewById(R.id.laughImage);
        awwImage = (ImageView) findViewById(R.id.awwImage);
        daDumTssImage = (ImageView) findViewById(R.id.drumImage);
        cricketImage = (ImageView) findViewById(R.id.cricketImage);
    }

    public static void playMedia(MediaPlayer mediaPlayer, final ImageView imageView) {
        if (mediaPlayer.isPlaying()) {
            makeImageSmaller(imageView);
            mediaPlayer.pause();
        } else {
            makeImageBigger(imageView);
            mediaPlayer.start();
            mediaPlayer.setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
                @Override
                public void onCompletion(MediaPlayer mp) {
                    makeImageSmaller(imageView);
                }
            });
        }
    }

    public void daDum(View view) {
        playMedia(daDumMp, daDumTssImage);
    }

    public void cricket(View view) {
        playMedia(cricket, cricketImage);
    }

    public void fakeLaugh(View view) {
        playMedia(fakeLaugh, laughImage);
    }

    public void aww(View view) {
        playMedia(aww, awwImage);
    }

    public static void makeImageBigger(ImageView imageView) {

        if (fade_out == null) {
            fade_out = new ScaleAnimation(1f, 1.4f, 1f, 1.4f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
            fade_out.setDuration(200);
            fade_out.setFillAfter(true);
        }

        imageView.startAnimation(fade_out);
    }

    public static void makeImageSmaller(ImageView imageView) {

        if (fade_in == null) {
            fade_in = new ScaleAnimation(1.4f, 1f, 1.4f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
            fade_in.setDuration(200);
            fade_in.setFillAfter(true);
        }

        imageView.startAnimation(fade_in);
    }
}

File: app/src/main/java/com/doapps/dojoke/MainActivity.kt
package com.doapps.dojoke

import android.app.Activity
import android.media.MediaPlayer
import android.os.Bundle
import android.view.View
import android.view.animation.Animation
import android.view.animation.ScaleAnimation
import android.widget.ImageView

@Suppress("UNUSED_PARAMETER")
class MainActivity : Activity() {
    private lateinit var daDumMp: MediaPlayer
    private lateinit var cricket: MediaPlayer
    private lateinit var aww: MediaPlayer
    private lateinit var fakeLaugh: MediaPlayer

    private lateinit var laughImage: ImageView
    private lateinit var awwImage: ImageView
    private lateinit var cricketImage: ImageView
    private lateinit var daDumTssImage: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Media players
        daDumMp = MediaPlayer.create(this, R.raw.badum)
        cricket = MediaPlayer.create(this, R.raw.cricket)
        aww = MediaPlayer.create(this, R.raw.aww)
        fakeLaugh = MediaPlayer.create(this, R.raw.laugh)

        // Images
        laughImage = findViewById<View>(R.id.laughImage) as ImageView
        awwImage = findViewById<View>(R.id.awwImage) as ImageView
        daDumTssImage = findViewById<View>(R.id.drumImage) as ImageView
        cricketImage = findViewById<View>(R.id.cricketImage) as ImageView
    }

    fun daDum(view: View) = playMedia(daDumMp, daDumTssImage)
    fun cricket(view: View) = playMedia(cricket, cricketImage)
    fun fakeLaugh(view: View) = playMedia(fakeLaugh, laughImage)
    fun aww(view: View) = playMedia(aww, awwImage)

    companion object {

        private val fadeOut = ScaleAnimation(1f, 1.4f, 1f, 1.4f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
        private val fadeIn = ScaleAnimation(1.4f, 1f, 1.4f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)

        fun playMedia(mediaPlayer: MediaPlayer, imageView: ImageView) {
            if (mediaPlayer.isPlaying) {
                makeImageSmaller(imageView)
                mediaPlayer.pause()
            } else {
                makeImageBigger(imageView)
                mediaPlayer.start()
                mediaPlayer.setOnCompletionListener { makeImageSmaller(imageView) }
            }
        }

        fun makeImageBigger(imageView: ImageView) {
            fadeOut.duration = 200
            fadeOut.fillAfter = true
            imageView.startAnimation(fadeOut)
        }

        fun makeImageSmaller(imageView: ImageView) {
            fadeIn.duration = 200
            fadeIn.fillAfter = true
            imageView.startAnimation(fadeIn)
        }
    }

    override fun onDestroy() {
        daDumMp.stop()
        cricket.stop()
        aww.stop()
        fakeLaugh.stop()
        super.onPause()
    }
}
